test/test_exact.cpp

```diff
@@ -87,3 +87,67 @@ TEST_P(ExactTest, QubitTriangle) {
 	SUCCEED() << "Mapping successful";
 }
 
+TEST_P(ExactTest, CommanderEncoding) {
+	MappingSettings settings{};
+	settings.encoding = 1; 
+	settings.grouping = 3;
+	IBMQ_Yorktown_mapper.map(settings);
+	IBMQ_Yorktown_mapper.dumpResult(GetParam() + "_exact_yorktown_commander.qasm");
+	IBMQ_Yorktown_mapper.printResult(std::cout, true);
+	SUCCEED() << "Mapping successful";
+	settings.encoding = 1; 
+	settings.grouping = 2;
+	IBMQ_Yorktown_mapper.map(settings);
+	IBMQ_Yorktown_mapper.dumpResult(GetParam() + "_exact_yorktown_commander.qasm");
+	IBMQ_Yorktown_mapper.printResult(std::cout, true);
+	SUCCEED() << "Mapping successful";
+	settings.encoding = 1; 
+	settings.grouping = -1;
+	IBMQ_Yorktown_mapper.map(settings);
+	IBMQ_Yorktown_mapper.dumpResult(GetParam() + "_exact_yorktown_commander.qasm");
+	IBMQ_Yorktown_mapper.printResult(std::cout, true);
+	SUCCEED() << "Mapping successful";
+	settings.encoding = 1; 
+	settings.grouping = -2;
+	IBMQ_Yorktown_mapper.map(settings);
+	IBMQ_Yorktown_mapper.dumpResult(GetParam() + "_exact_yorktown_commander.qasm");
+	IBMQ_Yorktown_mapper.printResult(std::cout, true);
+	SUCCEED() << "Mapping successful";
+}
+
+TEST_P(ExactTest, BimanderEncoding) {
+	MappingSettings settings{};
+	settings.encoding = 2; 
+	settings.grouping = 3;
+	IBMQ_Yorktown_mapper.map(settings);
+	IBMQ_Yorktown_mapper.dumpResult(GetParam() + "_exact_yorktown_commander.qasm");
+	IBMQ_Yorktown_mapper.printResult(std::cout, true);
+	SUCCEED() << "Mapping successful";
+	settings.encoding = 2; 
+	settings.grouping = 2;
+	IBMQ_Yorktown_mapper.map(settings);
+	IBMQ_Yorktown_mapper.dumpResult(GetParam() + "_exact_yorktown_commander.qasm");
+	IBMQ_Yorktown_mapper.printResult(std::cout, true);
+	SUCCEED() << "Mapping successful";
+	settings.encoding = 2; 
+	settings.grouping = -1;
+	IBMQ_Yorktown_mapper.map(settings);
+	IBMQ_Yorktown_mapper.dumpResult(GetParam() + "_exact_yorktown_commander.qasm");
+	IBMQ_Yorktown_mapper.printResult(std::cout, true);
+	SUCCEED() << "Mapping successful";
+	settings.encoding = 2; 
+	settings.grouping = -2;
+	IBMQ_Yorktown_mapper.map(settings);
+	IBMQ_Yorktown_mapper.dumpResult(GetParam() + "_exact_yorktown_commander.qasm");
+	IBMQ_Yorktown_mapper.printResult(std::cout, true);
+	SUCCEED() << "Mapping successful";
+}
+
+TEST_P(ExactTest, BDDLimits) {
+	MappingSettings settings{};
+	settings.bddLimits = 1;
+	IBMQ_Yorktown_mapper.map(settings);
+	IBMQ_Yorktown_mapper.dumpResult(GetParam() + "_exact_yorktown_commander.qasm");
+	IBMQ_Yorktown_mapper.printResult(std::cout, true);
+	SUCCEED() << "Mapping successful";
+}
```

Comment:
These tests do not cover all of the changes in this PR, e.g., they only test on a single bidirectional architecture.
Consider writing more tests. 
(CI should pass again now)